src/main/java/com/selfxdsd/storage/SelfProjects.java

```diff
@@ -142,70 +127,56 @@ public Project register(
     public Projects assignedTo(final int projectManagerId) {
         return new PmProjects(
             projectManagerId,
-            () -> {
-                final List<Project> assigned = new ArrayList<>();
-                final Result<Record> result = this.database.connect().jooq()
-                    .select()
-                    .from(SLF_PROJECTS_XDSD)
-                    .join(SLF_USERS_XDSD)
-                    .on(
-                        SLF_USERS_XDSD.USERNAME.eq(SLF_PROJECTS_XDSD.USERNAME)
-                            .and(
-                                 SLF_USERS_XDSD.PROVIDER.eq(
-                                     SLF_PROJECTS_XDSD.PROVIDER
-                                 )
+            () -> this.database.connect().jooq()
+                .select()
+                .from(SLF_PROJECTS_XDSD)
+                .join(SLF_USERS_XDSD)
+                .on(
+                    SLF_USERS_XDSD.USERNAME.eq(SLF_PROJECTS_XDSD.USERNAME)
+                        .and(
+                            SLF_USERS_XDSD.PROVIDER.eq(
+                                SLF_PROJECTS_XDSD.PROVIDER
                             )
-                    ).join(SLF_PMS_XDSD)
-                    .on(
-                        SLF_PROJECTS_XDSD.PMID.eq(SLF_PMS_XDSD.ID)
-                    )
-                    .where(
-                        SLF_PROJECTS_XDSD.PMID.eq(projectManagerId)
-                    )
-                    .fetch();
-                for(final Record rec : result) {
-                    assigned.add(this.projectFromRecord(rec));
-                }
-                return assigned.stream();
-            }
-        );
+                        )
+                ).join(SLF_PMS_XDSD)
+                .on(
+                    SLF_PROJECTS_XDSD.PMID.eq(SLF_PMS_XDSD.ID)
+                )
+                .where(
+                    SLF_PROJECTS_XDSD.PMID.eq(projectManagerId)
+                )
+                .stream()
+                .map(this::projectFromRecord));
     }
 
     @Override
     public Projects ownedBy(final User user) {
```

Comment:
@amihaiemil 
Well, the old code was loading the whole `ResultSet` into memory, cause `fetch()` is basically a collector. So before applying the `skip/limit` from filter classes, it would collect all the records + projects into memory (without pagination).
The fix was just to have a single stream by applying the `map` transfomer.
Still not the best solution, `offset/limit` should be  preferred since it is applied to the native query, while `skip/limit` is just filtering the whole result set stream (meaning that most of mapping `this::projectFromRecord` from upstream is useless).

Problem is that `skip/limit` is in downstream encapsulated into filter classes. 
Best order should be `resultset-stream` ->`skip/limit`->`map(this::projectFromRecord)` or even better `offset/limit`->`resultset-stream`->`map(this::projectFromRecord)`-> pass to filter class as stream supplier

Right now only `SelfProject#iterator` can take advantage of `offset/limit` (see how `this::projectFromRecord` is applied after `offset/limit` ).